### ISSUE TITLE
Remove explicit runtime dependency on sqlite3

### ISSUE THREAD

**Issue by liquid** [TryGhost/Ghost#6282]
I am installing Ghost in a production environment where I'm planning on using Postgresql. However when I'm running npm install, it fails due to a few things (I don't want to get into those in detail but it basically prevents me from deploying Ghost).

I understand that sqlite3 is the preferred default to get up-to-speed quickly but I'd like to see whether we could remove an explicit dependency on the database adapter and let the user decide which one they will go for. I am coming from a development background where environment specific database adapters are a working thing and found this to be very useful.

I understand that this may require some extra work and I'm happy to try to even approach this (although I'd not consider myself a nodejs ecosystem pro) but first of all I'd like to see what the scale of the work would be and whether if someone could help explaining the necessary changes this involves.

Provided this is doable, I may even try to take on this if there's someone who could help mentoring me, finding my way around the codebase.

Thank you in advance.

**Comment by ErisDS** [comment]
Hi @liquid, sorry to hear you're having trouble. I've read your post on the [sqlite3](https://github.com/mapbox/node-sqlite3/issues/584) repo & seen some comments in slack as well.

The reason the sqlite3 dependency is not optional (where the mysql & postgres deps are) is that sqlite3 is the no-config-required default. For the majority of installs, the user installs Ghost, gets an sqlite3 binary copied down for them, and everything *just works*. The path of least resistance here is Ubuntu & sqlite3.

Opting to use FreeBSD & postgres is the choice to use technologies which are harder to work with. There may be benefits, but this is not a common option and certainly not one that needs to be optimised for in my opinion.

If the sqlite3 dependency became optional, this would force the majority of our users to have a harder install process with extra config steps, so that a few people like yourself have a slightly easier time. Meanwhile, those people choosing to use a combo like FreeBSD & postgres are more likely to be technical and comfortable with extra steps, whereas the users who just want to install and go are often the less technical users.

In short, I don't think what you're proposing is a sensible optimisation for our user base. However, you can always raise it as a feature request on http://ideas.ghost.org/ to see if it gets traction.

**Comment by liquid** [comment]
Dear Hannah,

Thank you for the response. I appreciate the thorough explanation. Would you be able to elaborate on this part please?

> If the sqlite3 dependency became optional, this would force the majority of our users to have a harder install process with extra config steps

Am I right in thinking that this would only add another `npm install sqlite3` command to the install process or there's a much more complicated issue that isn't immediately obvious to me?

Cheers,

-- Attila

**Comment by ErisDS** [comment]
> Am I right in thinking that this would only add another npm install sqlite3 command 

This is exactly the problem. An extra step in the install process is a huge change, not to be taken lightly. Not only does it add a step, it's a step that you have to know about - running just `npm install` is common knowledge for node, this step would have to be documented. Additionally, every existing install guide would be incorrect, causing frustration and lots of people needing support.
